### DISCUSSION TITLE
How to connect with custom Json RPC?

### DISCUSSION THREAD

**Discussion by vrajdesai78** [wevm/wagmi#1792] [Q&A]
Hey, I want to connect with 5ire chain which is an EVM compatible L1. While using JsonRpcProvider, I need to mentioned the chain, but I am unable to do so. According to below example

![image](https://user-images.githubusercontent.com/43074241/216778575-e62d3123-c8fa-47ab-8d7f-563f74e76eb6.png)

I need to write Chain before configuring custom JsonRpcProvider, now without chain how can I do that, can anyone guide me. I want to use it in Next.js (typescript). 

Details for 5ire Chain:

```
Network Name: 5ireChain
New RPC Url: https://chain-node.5ire.network
Chain ID: 997
Currency Symbol: 5ire
Explorer URL: https://explorer.5ire.network
```

**Comment by tmm** [comment]
You are welcome to [submit a pull request](https://github.com/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md#chains) adding this chain or you can [define a custom Chain](https://wagmi.sh/react/chains#build-your-own) for your app.

**Comment by haidarabdillah** [comment]
this is really usefull finding on google is so hard
